### PR TITLE
Continuation on block production perf

### PR DESCRIPTION
**Pull request by tbenr**
Addressed some comments on previous PR and added a threshold config param

related to #7770 and #5808

## Documentation

- [ ] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [ ] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: teku/src/main/java/tech/pegasys/teku/cli/options/MetricsOptions.java [hunk 1]
```diff
@@ -119,6 +119,18 @@ public class MetricsOptions {
   private boolean blockProductionPerformanceEnabled =
       MetricsConfig.DEFAULT_BLOCK_PRODUCTION_PERFORMANCE_ENABLED;
 
+  @Option(
+      names = {"--Xmetrics-block-production-timing-tracking-warning-threshold"},
+      hidden = true,
+      showDefaultValue = Visibility.ALWAYS,
+      paramLabel = "<INTEGER>",
+      description =
+          "Whether block production is considered slow if it takes longer than this threshold (in ms)",
```
Comment (zilm13) [inline teku/src/main/java/tech/pegasys/teku/cli/options/MetricsOptions.java]: I think description could be improved but let's pass it to english-speaker

Comment (tbenr) [reply teku/src/main/java/tech/pegasys/teku/cli/options/MetricsOptions.java]: The author is copilot :)

Comment (tbenr) [reply teku/src/main/java/tech/pegasys/teku/cli/options/MetricsOptions.java]: do you like the new version?

Comment (rolfyone) [reply teku/src/main/java/tech/pegasys/teku/cli/options/MetricsOptions.java]: not really...
```
The time (in ms) at which block production is to be considered 'slow'. If set to 100, block production taking at least 100ms would raise a warning.
```

Comment (tbenr) [reply teku/src/main/java/tech/pegasys/teku/cli/options/MetricsOptions.java]: updated

**Review by rolfyone — APPROVED** [review]
LGTM

**Pull request merged**